### ISSUE TITLE
parts of global_prefs get reported as variety in trickle up messages

### ISSUE THREAD

**Issue by ChristianBeer** [BOINC/boinc#1485]
I just found 26 entries in msg_from_host that accumulated over the last weeks that didn't have the correct variety of "cpu_time" instead they had the beginning of the global prefs XML or a something else stored (I deleted the newlines as they were messing with the layout).

```
| id     | create_time | hostid | variety                                                                                                                                                                                                                                                       | handled | xml                                                                                                                                                                                          |
| 433101 |  1435373824 |  33244 | <source_project>http://bam.boincstats.com/</source_project><mod_time>1406799579</mod_time>                                                                                                                                                                   |       0 | <result_name>cmsvm_GA-p[e30-50MB_Lin64f]_1_Oryza-sativa-Japonica-Group_CM000138.lin.EMBL_RF00177_SSU_rRNA_5_1330438623_83865_63</result_name><time>1435370520</time>         |
| 443986 |  1436319757 |  27843 | <source_project>http://bam.boincstats.com/</source_project><mod_time>1432043467</mod_time><venue name="New: 2010-12-07 07:24:28"><ram_max_used_idle_pct>0</ram_max_used_idle_pct><work_buf_min_days>0.01</work_buf_min_days><cpu_usage_limit>100</cpu_us |       0 |       <result_name>cmsvm2_GA-p[e30-50MB_Lin64f]_1_Anopheles-gambiae-str.PEST_CM000358.lin.EMBL_RF00028_Intron_gpI_1330438623_1400_14</result_name>      <time>1436319500</time>         |
| 447532 |  1436637105 |  13748 | <source_project>http://gerasim.boinc.ru/</source_project>    <source_scheduler>http://gerasim.boinc.ru/sched</source_scheduler>    <mod_time>1404481426</mod_time>   <cpu_scheduling_period_minutes>120</cpu_scheduling_period_minutes>    <idle_time_to |       0 |       <result_name>cmsvm2_GA-p[e20-30MB_Lin64f]_1_Oryza-sativa-Japonica-Group_AP008214.lin.EMBL_RF00028_Intron_gpI_1349111823_52164_33</result_name>      <time>1436636371</time>        |
| 451898 |  1436980689 |  35203 | <source_project>http://bam.boincstats.com/</source_project><mod_time>1427008505</mod_time><venue name="home"><ram_max_used_idle_pct>90</ram_max_used_idle_pct><run_on_batteries/><work_buf_min_days>0.2</work_buf_min_days><cpu_usage_limit>100</cpu_us |       0 |       <result_name>cmsvm_GA-p[e20-30MB_Lin64f]_1_Drosophila-melanogaster-(fruit-fly)_AE014297.lin.EMBL_RF00028_Intron_gpI_1349111823_13748_13</result_name>      <time>1436977561</time> |
| 478215 |  1439012242 |  36761 | windows_intelx86                                                                                                                                                                                                                                             |       0 |       <result_name>cmsvm_GA-p[e30-50MB_Lin64f]_1_Oryzias-latipes-(Japanese-medaka)_DG000007.lin.EMBL_RF00028_Intron_gpI_1330438623_94696_58</result_name>      <time>1439011888</time>   |
```

I don't know where this originates from but think this is another reason for a better XML parser like already discussed in #1470. It could be the client sending it wrong or the scheduler storing it wrong I didn't investigate.

**Comment by ChristianBeer** [comment]
I fixed this by changing [sched/trickle_handler.cpp#L69](https://github.com/BOINC/boinc/blob/master/sched/trickle_handler.cpp#L69) like this:
```cpp
        retval = handle_trickle(mfh);
        if (!retval) {
            mfh.handled = true;
            mfh.update();
        } else {
            mfh.handled = 10;
            mfh.update();
        }
```
This will not delete the messages but at least the trickle handler will not choke every time this happens.

**Comment by AenBleidd** [comment]
Looks like this has been already fixed in #2630
